library/src/main/java/io/constructor/core/ConstructorIo.kt

```diff
@@ -184,12 +184,12 @@ object ConstructorIo {
     /**
      * Tracks search result click events
      */
-    fun trackSearchResultClick(itemName: String, customerId: String, searchTerm: String = Constants.QueryConstants.TERM_UNKNOWN, sectionName: String? = null): Completable {
+    fun trackSearchResultClick(itemName: String, customerId: String, searchTerm: String = Constants.QueryConstants.TERM_UNKNOWN, sectionName: String? = null, resultID: String? = null): Completable {
         preferenceHelper.getSessionId(sessionIncrementHandler)
         val sName = sectionName ?: preferenceHelper.defaultItemSection
         return dataManager.trackSearchResultClick(itemName, customerId, searchTerm, arrayOf(
                 Constants.QueryConstants.AUTOCOMPLETE_SECTION to sName
-        ))
+        ), resultID)
```

Comment:
The existing system seems to be designed to pass optional parameters into an `encodedParams` array instead of adding extra params to the data manager calls